h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -215,34 +117,34 @@ public AutoMLBuildSpec getBuildSpec() {
   public Vec getFoldColumn() { return foldColumn; }
   public Vec getWeightsColumn() { return weightsColumn; }
 
-  private Frame trainingFrame;    // required training frame: can add and remove Vecs, but not mutate Vec data in place
-  private Frame validationFrame;  // optional validation frame; the training_frame is split automagically if it's not specified
-  private Frame blendingFrame;
-  private Frame leaderboardFrame; // optional test frame used for leaderboard scoring; if not specified, leaderboard will use xval metrics
+  public StepDefinition[] getTrainedSteps() { return trainedSteps; }
 
-  private Vec responseColumn;
-  private Vec foldColumn;
-  private Vec weightsColumn;
+  Frame trainingFrame;    // required training frame: can add and remove Vecs, but not mutate Vec data in place
+  Frame validationFrame;  // optional validation frame; the training_frame is split automagically if it's not specified
+  Frame blendingFrame;
+  Frame leaderboardFrame; // optional test frame used for leaderboard scoring; if not specified, leaderboard will use xval metrics
 
-  private Key<Grid> gridKeys[] = new Key[0];  // Grid key for the GridSearches
+  Vec responseColumn;
+  Vec foldColumn;
+  Vec weightsColumn;
 
-  private Date startTime;
-  private static Date lastStartTime; // protect against two runs with the same second in the timestamp; be careful about races
-  private Countdown runCountdown;
-  private Job job;                  // the Job object for the build of this AutoML.
+  Date startTime;
+  Countdown runCountdown;
+  Job<AutoML> job;                  // the Job object for the build of this AutoML.
+  WorkAllocations workAllocations;
+  StepDefinition[] trainedSteps;
```

Comment:
made those properties package private: `AutoML` instance is injected to `TrainingSteps` and `TrainingStep` so that each model definition can access as much information as possible: by being package private, they are available to the classes already mentioned, and subclasses can access them through delegate accessors.

I still don't want to make those properties public to avoid backwards compatibility issues with future changes.